ledger-core/testutils/journalinglogger/iterator.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package journalinglogger
+
+import (
+	"reflect"
+
+	"github.com/insolar/assured-ledger/ledger-core/conveyor/smachine"
+	"github.com/insolar/assured-ledger/ledger-core/testutils/debuglogger"
+)
+
+type JournalIterator struct {
+	id       int
+	await    chan struct{}
+	position int
+	journal  *SyncJournal
+}
+
+func (it *JournalIterator) WaitEvent(predicate func(debuglogger.UpdateEvent) bool) bool {
+	for {
+		event, ok := it.journal.Get(it.position)
+		if ok {
+			it.position++
+
+			if predicate(event) {
+				return true
+			}
+			continue
+		}
+
+		_, closed := <-it.await
+		if !closed {
+			return false
+		}
+	}
+}
+
+func (it *JournalIterator) WaitStop(sm smachine.StateMachine, count int) bool {
+	var declarationType = reflect.TypeOf(sm)
+
+	return it.WaitEvent(func(event debuglogger.UpdateEvent) bool {
+		if reflect.TypeOf(event.SM) == declarationType {
+			if event.Update.UpdateType == "stop" {
```

Comment:
and `event.Update.UpdateType == "error"`

Reply:
sure